godotproject/Player.gd

```diff
@@ -174,11 +176,37 @@ func _physics_process(delta):
 		else:
 			emit_signal("tongue_start", get_tongue_direction())
 
+func takeDamage(damageTaken):
+	if isVulnerable:
+		print("Player receiving %d damage" % damageTaken)
+		health -= damageTaken
+		print("Player took damage, has %d health +remaining" % health)
+		if health <= 0:
+			print("Player is now dead")
+			die()
+			
+		#After the player gets hit set the player to invulnerable to damage for 1 second
+		isVulnerable = false
+		
+		#sets up timer to trigger after 1 second and set player to vulnerable again
+		print("Timer has started")
+		var invulnerable_timer = Timer.new()
```

Comment:
Should this timer object be created each time the enemy takes damage? Or is there a more efficient way to go about it?